util/devel/look_for_calls.py

```diff
@@ -122,18 +122,35 @@ def check_for_calls(functions, search_dir, exclude_paths=None, rel_paths=True):
 
         found_calls = False
         for func in functions:
-            out = cscope_find_calls(func, cscope_database_name)
-            if out:
-                found_calls = True
-                log_error('found call to "{0}"'.format(func))
-                sys.stdout.write(out.replace(rel_dir, '') + '\n')
+          # If func is a tuple consider the first element to be the function
+          # we're searching for and the second an alternative to suggest
+          # to the user.
+          alternative = None
+          if isinstance(func, tuple):
+            alternative = func[1]
+            func = func[0]
+            
+          out = cscope_find_calls(func, cscope_database_name)
+          if out:
+              found_calls = True
+              msg = 'found call to "{0}"'.format(func)
+              if alternative is not None:
+                msg += ' consider using "{0}" instead'.format(alternative)
+              log_error(msg)
+              sys.stdout.write(out.replace(rel_dir, '') + '\n')
 
         return found_calls
 
 
 def get_alloc_funcs():
     """Return a list of the possible C alloc/dealloc routines"""
-    std = ['malloc', 'calloc', 'realloc', 'free']
+    # If list element is a tuple then consider the second element to
+    # be an alternative for the first. If the list element is a
+    # not a tuple then their is no known alternative.
```

Comment:
This comment and the comment you added near the top seem a bit redundant with each other. Maybe just keep one of them?

Reply:
Yeah, I don't really like that I having this information duplicated. 

If this was production code and not a simplish script used by our CI system I'd say we should use a variant and a struct with named fields to make it self-documenting, but that seems like more overhead then it's worth for a simple Python script.

Anyway, I want it to be easy for an impatient developer to figure out how to modify\add the suggestions and if they just grep for "malloc" (or whatever function they want to update) they'd see this comment and not the other one. So I think it's regrettable but worth duplicating the info.

Reply:
Makes sense